### PR TITLE
Refactored some of the lesson progress javascript and fixed the race …

### PR DESCRIPTION
**Pull request by captainhurst**
…condition causing the lessons to remain closed on first load

### CODE REVIEW

File: public/javascripts/context_modules.js [hunk 1]
```diff
@@ -203,40 +229,74 @@ import 'compiled/jquery.rails_flash_notifications'
...
         var courseItems = JSON.parse(window.localStorage.getItem("course_items"));
 
-        courseItems.map(function(item){
-          let progressBar = $("#"+item.id+" .sm-unit-header_progress-container .sm-unit-header_progress-bar");
-          let progressText = $("#"+item.id+" .sm-unit-header_progress-container .sm-progress-percentage");
-          let progress = calculateUnitProgress(item);
+        courseItems.map((unit) => {
+          let progressBar = $("#"+unit.id+" .sm-unit-header_progress-container .sm-unit-header_progress-bar");
+          let progressText = $("#"+unit.id+" .sm-unit-header_progress-container .sm-progress-percentage");
+          let progress = modules.calculateUnitProgress(unit);
```
Comment (yuric) [inline public/javascripts/context_modules.js]: was there a hick up with unit progress?"

Comment (captainhurst) [reply public/javascripts/context_modules.js]: No. Renamed a variable and incorporated 2 loops over the same dom elements into 1 loop.

Comment (yuric) [reply public/javascripts/context_modules.js]: Right. I was attempting to understand why. It looks like a minor refactor.

Comment (captainhurst) [reply public/javascripts/context_modules.js]: Beautiful!

**Pull request merged**